zetaclient/zetacore/background_threads.go

```diff
@@ -0,0 +1,162 @@
+package zetacore
+
+import (
+	"context"
+	"time"
+
+	"cosmossdk.io/errors"
+
+	"github.com/zeta-chain/zetacore/pkg/retry"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+	zctx "github.com/zeta-chain/zetacore/zetaclient/context"
+)
+
+// HandleTSSUpdate is a background thread that listens for TSS updates; it returns when the TSS address is updated
+func (c *Client) HandleTSSUpdate(ctx context.Context) error {
+	app, err := zctx.FromContext(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to get app context")
+	}
+
+	logger := app.Logger().With().Str("module", "HandleTSSUpdate").Logger()
+
+	// Initial TSS retrieval
+	tss, err := retry.DoTypedWithBackoffAndRetry[observertypes.TSS](func() (observertypes.TSS, error) {
+		return c.GetTSS(ctx)
+	}, retry.DefaultConstantBackoff())
+	if err != nil {
+		logger.Warn().Err(err).Msg("unable to get initial tss")
+		return err
+	}
+
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			{
+				tssNew, err := c.GetTSS(ctx)
+				if err != nil {
+					logger.Warn().Err(err).Msg("unable to get new tss")
+					continue
+				}
+
+				if tssNew.TssPubkey == tss.TssPubkey {
+					continue
+				}
+				logger.Info().Msgf("tss address is updated from %s to %s", tss.TssPubkey, tssNew.TssPubkey)
+				tss = tssNew
```

Comment:
what is the use of this line? Not sure if I'm missing something but we don't return `tss` so this line seems to not be doing anything.